app/src/main/java/be/huffle/allowanceplanner/activities/AddExpenseActivity.java

```diff
@@ -94,22 +94,56 @@ public void onClick(DialogInterface dialog, int id)
 		alert.show();
 	}
 
-	private void addExpense(float expense, String description)
+	private void showDialogWithOptions(String message, float balance, float expense)
+	{
+		boolean canAddExpense;
+		AlertDialog.Builder builder = new AlertDialog.Builder(this);
+
+		builder.setMessage(message);
+		builder.setPositiveButton("Yes",
+				new DialogInterface.OnClickListener()
+				{
+					@Override
+					public void onClick(DialogInterface dialog, int id)
+					{
+						dialog.dismiss();
+						addExpense(balance, expense);
+					}
+				});
+		builder.setNegativeButton("No",
+				new DialogInterface.OnClickListener()
+				{
+					@Override
+					public void onClick(DialogInterface dialog, int id)
+					{
+						dialog.dismiss();
+					}
+				});
+		AlertDialog alert = builder.create();
+		alert.show();
+	}
+
+	private void allowedToAdd(float expense, String description)
```

Comment:
This name has a bit of an odd name. From it, I would expect it to return a boolean which indicates whether or not it's allowed to add an expense. However, that does not seem to be the case.

Perhaps something like `tryAddExpense` would be better?